e2e/specs/st_multiselect.spec.js

```diff
@@ -212,37 +187,109 @@ describe("st.multiselect", () => {
                 "value 7: []" +
                 "value 8: []" +
                 "value 9: []" +
+                "value 10: []" +
+                "value 11: []" +
                 "multiselect changed: False"
             );
         });
       });
     });
 
     it("calls callback if one is registered", () => {
-      cy.get(".stMultiSelect")
-        .should("have.length", 9)
-        .last()
+      cy.getIndexed(".stMultiSelect", 10)
         .find("input")
         .click();
       cy.get("li")
         .first()
         .click();
 
       cy.get("[data-testid='stText']")
-        .should("have.length", 10)
         .should(
-          "have.text",
-          "value 1: []" +
-            "value 2: ['female']" +
-            "value 3: []" +
-            "value 4: ['tea', 'water']" +
-            "value 5: []" +
-            "value 6: []" +
-            "value 7: []" +
-            "value 8: []" +
-            "value 9: ['male']" +
+          "contain.text",
+            "value 11: ['male']" +
             "multiselect changed: True"
         );
     });
   });
+
+  describe("when using max_selections for st.multiselect", () => {
+    beforeEach(() => {
+      // this is the 'close button' element for 'Male'
+      cy.get('.stMultiSelect span[data-baseweb="tag"] span:last-child')
+        .eq(0)
+        .click();
+    });
+    it("should show the correct text when maxSelections is reached not within a form and closing input", () => {
+      selectNthOptionForKthMultiselect(0, 8)
+      cy.getIndexed(".stMultiSelect", 8).then(el => {
```

Comment:
(Also applies to a bunch of tests below): Is there a reason a `.then()` gets chained here instead of just calling `getIndexed(...).find("input")...`?

If it's to make sure the multiselect rerenders after the kth element is clicked on, this probably isn't how we want to do this. If this is indeed why it's being done, I think it's just working by coincidence because the `.then()` handler gets put on the event loop so that the browser has time to do the rerender before it gets run.

Not sure if there's a strict rule against this, but in general, we've been avoiding explicitly working with promises in our cypress tests when possible. Instead, the general way to do this would be to wait for the just-selected item to appear in the multiselect before checking for the max selected text.